src/transformStateWithClones.js

```diff
@@ -65,7 +65,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const resultingArray = [];
+  const objTmp = { ...state };
+
+  for (const arrayItem of transforms) {
+    if (arrayItem.operation === 'addProperties') {
+      for (const addprops in arrayItem.properties) {
```

Comment:
```suggestion
      for (const property in transformation.properties) {
```
When I read 'addprops in arrayItem.properties', I can't understand anything about these variables. Give names that help understand what these variables contain